src/components/TaxPayer/SpouseAndDependent.test.tsx

```diff
@@ -0,0 +1,250 @@
+    screen.getByText('SSN / TIN')
+
+    screen.getByRole('button', {
+      name: /Add/
+    })
+
+    screen.getByRole('button', {
+      name: /Close/
+    })
+  })
+  it('saves and edits a spouse', async () => {
+    render(
+    <Provider store={store}>
+      <SpouseInfo />
+    </Provider>
+    )
+
+    const addButton = screen.getByRole('button', {
+      name: /Add/
+    })
+
+    fireEvent.click(addButton)
+
+    const createButton = screen.getByRole('button', {
+      name: /Add/
+    })
+
+    screen.getByRole('button', {
+      name: /Close/
+    })
+
+    // get all three text inputs
+    const inputs = screen.getAllByRole('textbox')
+    const [firstNameInput, lastNameInput, ssidInput] = inputs
+
+    // add values for each input
+    userEvent.type(firstNameInput, 'Sally K')
+    userEvent.type(lastNameInput, 'Ride')
+    fireEvent.change(ssidInput, { target: { value: '123456789' } })
+
+    // click the create button
+    fireEvent.click(createButton)
+
+    // expect first and last name to be concatenated
+    await screen.findByText('Sally K Ride')
+    // expect ssid to appear without hyphens
+    screen.getByText('123456789')
+
+    const editButton = screen.getByLabelText('edit')
+
+    // click the edit button
+    fireEvent.click(editButton)
+
+    // get all the inputs again, which should be filled this time
+    const filledInputs = await screen.findAllByRole('textbox')
+    const [filledFirstName, filledLastName, filledSsid] = filledInputs as HTMLInputElement[]
+
+    // expect the edit button to no longer be in the document
+    expect(editButton).not.toBeInTheDocument()
+
+    // assert that the input values match what was entered
+    expect(filledFirstName.value).toBe('Sally K')
+    expect(filledLastName.value).toBe('Ride')
+    expect(filledSsid.value).toBe('123-45-6789')
+
+    // delete the old values and add new ones
+    userEvent.type(filledFirstName, '{selectall}{del}Fella')
+    userEvent.type(filledLastName, '{selectall}{del}McGee')
+    fireEvent.change(filledSsid, { target: { value: '987654321' } })
+
+    // wait for redux-persist to do some async stuff
+    await waitFor(() => {})
+
+    // click the add button to save the new values
+    fireEvent.click(screen.getByRole('button', {
+      name: /Add/
+    }))
+
+    // expect the new names to be concatenated and new ssid to appear without hyphens
+    await screen.findByText('Fella McGee')
+    screen.getByText('987654321')
+
+    const deleteButton = screen.getByLabelText('delete')
+
+    // click the delete button
+    fireEvent.click(deleteButton)
+
+    // the add button is back
+    screen.getByRole('button', {
+      name: /Add/
+    })
+
+    const inputsAfterDelete = screen.queryAllByRole('textbox')
+
+    // expect input fields to not be in the document
+    expect(inputsAfterDelete).toHaveLength(0)
+  })
+  it('does not save when required fields not completed', async () => {
+    render(
+    <Provider store={store}>
```

Comment:
this indentation seems off, not a big deal but there several instances.

Reply:
Yeah, I don't know what's happening with my editor throwing that off. I think once this is in I'd like to raise a separate PR to add a `prettier` step to the pipeline so mal-formatted code will never reach GitHub